src/mode/mode_struct.rs

```diff
@@ -1183,6 +1192,52 @@ impl<T: Transformation> Mode<T> {
         v
     }
 
+    fn is_in_sync(
+        &self,
+        jump_max: UnitValue,
+        time_expired: bool,
+        prev_value: UnitValue,
+        current_value: UnitValue,
+        target_value: UnitValue,
+        in_sync: bool,
+    ) -> bool {
+
+        let current_distance: f64 = current_value.get() - target_value.get();
+        let prev_distance: f64 = prev_value.get() - target_value.get();
+        let same_side: bool = (current_distance < 0.0 && prev_distance < 0.0) ||
+                              (current_distance > 0.0 && prev_distance > 0.0);
+        let approaching_target: bool = same_side && (current_distance.abs() < prev_distance.abs());
+
+        // If we were in sync less than CONTROL_MOVE_TIMEOUT ms ago, stay in sync
+	    if in_sync && !time_expired {
+            return true;
+        }
+
+        // If Jump Max = 100% then we are in sync by definition. (movements less than Jump
+        // Min will be handled downstream of this)
+        if jump_max.is_one() {
+            return true;
+        }
+
+        // New takeover logic. Goals:
+        //
+        // 1) Avoid awkward "backwards" jumps when approaching the target.
+        // 2) Ensure that we pick up fast movements that *pass through* the target, even if
+        //    the individual controller values never actually fall inside the Jump Max range.
+        // 3) Enable a typical "set and forget" Jump Max of 3% or less without sync loss.
+        //
+        // Inspired by controllers/softtakeover.cpp in the Mixxx DJ project.
+
+        if current_distance.abs() <= jump_max.get() && !approaching_target {
+            return true;
+        } else if !same_side && !time_expired {
```

Comment:
This is probably a matter of taste, but since you have used "early return" everywhere else, it would be consequent to skip the `else` here as well and then just write the following `if` block in the next line.